bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/soap/SOAPConnector.java

```diff
@@ -118,19 +122,46 @@ private Request prepareSOAPRequest(SCPDServiceType service, String soapAction, M
     }
 
     /**
-     * execute a SOAP request
+     * execute a SOAP request with cache
      *
-     * @param service the service to send the action to
-     * @param soapAction the action itself
-     * @param arguments arguments to send along with the request
+     * @param soapRequest the request itself
      * @return the SOAPMessage answer from the remote host
      * @throws Tr064CommunicationException if an error occurs during the request
      */
-    public synchronized SOAPMessage doSOAPRequest(SCPDServiceType service, String soapAction,
-            Map<String, String> arguments) throws Tr064CommunicationException {
+    public SOAPMessage doSOAPRequest(SOAPRequest soapRequest) throws Tr064CommunicationException {
         try {
-            Request request = prepareSOAPRequest(service, soapAction, arguments).timeout(SOAP_TIMEOUT,
-                    TimeUnit.MILLISECONDS);
+            SOAPMessage soapMessage = Objects.requireNonNull(soapMessageCache.putIfAbsentAndGet(soapRequest, () -> {
```

Comment:
How are you making sure that requests that modify remote state aren't getting cached? Are all requests getters?